src/jshint.js

```diff
@@ -4225,7 +4225,7 @@ var JSHINT = (function() {
     var i = 0;
     var inof = ["in", "of"];
     var level = 0; // BindingPattern "level" --- level 0 === no BindingPattern
-    var comma; // First comma punctuator at level 0
+    var firstcomma; // First comma punctuator at level 0
```

Comment:
Since it was on a private repository that we analyzed on production I don't have access to the actual code that was causing this.

But you can see the offending call to the `comma` function https://github.com/gtanner/jshint/blob/master/src/jshint.js#L4362 that can crash before the rename of the local var to `firstcomma`.